src/witan/gateway/components/kafka.clj

```diff
@@ -0,0 +1,77 @@
+(ns witan.gateway.components.kafka
+  (:require [witan.gateway.protocols    :refer [SendMessage]]
+            [com.stuartsierra.component :as component]
+            [taoensso.timbre            :as log]
+            [cheshire.core              :as json]
+            [clj-kafka.producer         :as kafka]
+            [clj-kafka.zk               :as zk]
+            [clj-kafka.consumer.zk      :as kafka-zk]
+            [clj-kafka.core             :as kafka-core]
+            [clojure.core.async         :as async :refer [go-loop chan <! close! put!]]))
+
+(defrecord KafkaProducer [host port]
+  SendMessage
+  (send-message! [component topic raw-message]
+    (let [message (json/generate-string raw-message)]
+      (if-let [{:keys [connection]} component]
+        (if-let [error (kafka/send-message connection (kafka/message (name topic) (.getBytes message)))]
+          (log/error "Failed to send message to Kafka:" error)
```

Comment:
I wonder if we could handle this better. ATM the connection could be broken and we'd just drop all the messages w/o feeding back to the client process IIUC